src/Workspaces/Core/Portable/Serialization/SerializableSourceText.cs

```diff
@@ -81,5 +119,39 @@ public static ValueTask<SerializableSourceText> FromTextDocumentStateAsync(TextD
                     cancellationToken);
             }
         }
+
+        public void Serialize(ObjectWriter writer, SolutionReplicationContext context, CancellationToken cancellationToken)
+        {
```

Comment:
moved inside so we don't have to leak out our internal state.

Reply:
This is much nicer.